packages/arb-ts/src/lib/l1Bridge.ts

```diff
@@ -102,82 +109,60 @@ export class L1Bridge {
-          this.l1Signer
-        )
-        const [balance] = await ethERC20TokenContract.functions.balanceOf(
-          walletAddress
-        )
+    if (!this.contractExists(erc20L1Address))
+      throw new Error(`No ERC20 at ${erc20L1Address} `)
 
-        const [allowance] = await ethERC20TokenContract.functions.allowance(
-          walletAddress,
-          gatewayAddress
-        )
-        // non-standard
-        const symbol = await ethERC20TokenContract.functions
-          .symbol()
-          .then(([res]) => res)
-          .catch(() => addressToSymbol(erc20L1Address))
+    // If this will throw if not an ERC20, which is what we *want*.
+    const ethERC20TokenContract = await ERC20__factory.connect(
+      erc20L1Address,
+      this.l1Signer
+    )
+    const [balance] = await ethERC20TokenContract.functions.balanceOf(
+      walletAddress
+    )
 
-        const decimals = await ethERC20TokenContract.functions
-          .decimals()
-          .then(([res]) => res)
-          .catch(() => 18)
+    const [allowance] = await ethERC20TokenContract.functions.allowance(
+      walletAddress,
+      gatewayAddress
+    )
+    // non-standard
+    const symbol = await ethERC20TokenContract.functions
+      .symbol()
+      .then(([res]) => res)
+      .catch(() => addressToSymbol(erc20L1Address))
 
-        const name = await ethERC20TokenContract.functions
-          .name()
-          .then(([res]) => res)
-          .catch(() => symbol + '_Token')
-        const allowanceLimit = BigNumber.from(
-          '0xffffffffffffffffffffffff'
-        ) /** for ERC20s that cap approve at 96 bits  */
-        const allowed = await allowance.gte(allowanceLimit.div(2))
-        tokenData.ERC20 = {
-          contract: ethERC20TokenContract,
-          balance,
-          allowed,
-          symbol,
-          decimals,
-          name,
-        }
-      } else {
-        throw new Error(`No ERC20 at ${erc20L1Address} `)
-      }
-    } else {
-      const ethERC20TokenContract = await ERC20__factory.connect(
-        erc20L1Address,
-        this.l1Signer
-      )
-      const [balance] = await ethERC20TokenContract.functions.balanceOf(
-        walletAddress
-      )
-      tokenData.ERC20.balance = balance
+    const decimals = await ethERC20TokenContract.functions
+      .decimals()
+      .then(([res]) => res)
+      .catch(() => 18)
 
-      if (!tokenData.ERC20.allowed) {
-        const [allowance] = await ethERC20TokenContract.functions.allowance(
-          walletAddress,
-          gatewayAddress
-        )
-        tokenData.ERC20.allowed = allowance.gte(MIN_APPROVAL.div(2))
-      }
+    const name = await ethERC20TokenContract.functions
+      .name()
+      .then(([res]) => res)
+      .catch(() => symbol + '_Token')
+    const allowanceLimit = BigNumber.from(
+      '0xffffffffffffffffffffffff'
+    ) /** for ERC20s that cap approve at 96 bits  */
+    const allowed = await allowance.gte(allowanceLimit.div(2))
```

Comment:
Doesn't seem like this should need an await?

Reply:
oops